File: TDDMicroExercises.ProposedSolution/CSharp/TirePressureMonitoringSystem.Tests/StubSensor.cs

using System.Collections.Generic;

namespace TDDMicroExercises.OneSolution.TirePressureMonitoringSystem.Tests
{
    public class StubSensor : ISensor
    {
        private readonly Stack<double> _returnedValues = new Stack<double>();

        public void StubCallToPopNextPressurePsiValue(double returnedValue) 
        {
            StubCallToPopNextPressurePsiValues(new double[] { returnedValue });
        }

        public void StubCallToPopNextPressurePsiValues(double[] returnedValues)
        {
            foreach (double expectedValue in returnedValues)
            {
                _returnedValues.Push(expectedValue);                 
            }
        }

        double ISensor.PopNextPressurePsiValue()
        {
            return _returnedValues.Pop();
        }
    }
}

File: TDDMicroExercises.ProposedSolution/Java/TirePressureMonitoringSystem/src/test/java/tddmicroexercises/tirepressuremonitoringsystem/StubSensor.java
package tddmicroexercises.tirepressuremonitoringsystem;

import java.util.Stack;

public class StubSensor implements ISensor {

	private Stack<Double> values = new Stack<Double>();

	public double popNextPressurePsiValue() {
		return values.pop();
	}

	public void pushPsiValue(double d) {
		values.push(d);
		
	}

}
